agent/config/runtime_test.go

```diff
@@ -7194,11 +7196,11 @@ func TestSanitize(t *testing.T) {
 				"Enabled": false,
 				"AllowReuse": false,
 				"AuthMethod": {
-					"ACLAuthMethodEnterpriseFields": {},
+					"ACLAuthMethodEnterpriseFields": ` + entAuthMethodFieldsJSON + `,
```

Comment:
These tests couldn't pass for both oss and enterprise. Not sure how these could have passed before without these changes. The AutoConf config builder creates the AuthMethod [here](https://github.com/hashicorp/consul-enterprise/blob/5ff983fd34d7d643ee9b4083508d5a8af7512b50/agent/config/builder.go#L2261-L2280)

The EnterpriseMeta is always a concrete struct so it will always end up in here for enterprise but blank for OSS. As for the auth method enterprise fields I think it should be the same way due to the ent fields being embedded as a concrete instance in the ACLAuthMethod struct.